### PR TITLE
Add Walmart med price

### PR DESCRIPTION
**Pull request by jknapp25**
For the Walmart pilot we're surface med prices if the prescribed meds are in a select list. This will only surface a price if all the medications in the order are present in the list.

Example with 2 meds ($4 + $10) in the list:

image

### CODE REVIEW

File: apps/patient/src/components/PharmacyCard.tsx [hunk 1]
```diff
@@ -26,6 +26,7 @@ interface PharmacyCardProps {
   onSetPreferred?: () => void;
   selectable?: boolean;
   showDetails?: boolean;
+  price: number;
```
Comment (marisbest2) [inline apps/patient/src/components/PharmacyCard.tsx:29]: This should be optional I assume?

File: apps/patient/src/utils/general.ts [hunk 2]
```diff
@@ -207,3 +211,40 @@ export const convertReadyByToUTCTimestamp = (readyBy: string, readyByDay: string
...
+        (!med.form || fill.treatment.name.toLowerCase().includes(med.form)) &&
+        med.daysSupply === fill?.prescription?.daysSupply &&
+        med.dispenseQuantity === fill?.prescription?.dispenseQuantity
+    );
+
+    // If a matching fill is not found, set allFillsMatch to false
+    if (!match) {
+      allFillsMatch = false;
+      break;
+    }
```
Comment (marisbest2) [inline apps/patient/src/utils/general.ts:242]: Just return `null` here I think?

File: apps/patient/src/utils/general.ts [hunk 3]
```diff
@@ -207,3 +211,40 @@ export const convertReadyByToUTCTimestamp = (readyBy: string, readyByDay: string
...
+ */
+export const getTotalWalmartOrderPrice = (fills: types.Fill[]): number => {
+  let totalPrice = 0;
+  let allFillsMatch = true;
+
+  for (const fill of fills) {
+    // Check if the fill is in the walmartMedLookup list
+    const match = walmartMedLookup.find(
+      (med) =>
+        new RegExp(`${med.name}.*${med.strength}`, 'i').test(fill.treatment.name) &&
```
Comment (marisbest2) [inline apps/patient/src/utils/general.ts:231]: Can we not just do an exact match here? Theoretically worried about partial matches. For example

`Acetaminophen 500mg - Caffeine 60mg - Pyrilamine maleate 15mg` (excedrin)

would match `Acetaminophen 15mg` despite not being the same thing. (obviously a contrived example)

File: apps/patient/src/utils/general.ts [hunk 4]
```diff
@@ -207,3 +211,40 @@ export const convertReadyByToUTCTimestamp = (readyBy: string, readyByDay: string
...
+      allFillsMatch = false;
+      break;
+    }
+
+    // If a matching fill is found, add its price to the total price
+    totalPrice += match.price;
+  }
+
+  // If all fills match, return the total price; otherwise, return 0
+  return allFillsMatch ? totalPrice : 0;
```
Comment (marisbest2) [inline apps/patient/src/utils/general.ts:249]: `0` is a valid price. We should return `null` here if we can't figure out a price

File: apps/patient/src/views/Main.tsx [hunk 5]
```diff
@@ -21,6 +25,7 @@ interface OrderContextType {
   setOrder: (order: Order) => void;
   logo: any;
   isDemo: boolean;
+  totalWalmartPrice: number;
```
Comment (marisbest2) [inline apps/patient/src/views/Main.tsx:28]: this should be `number | null`

File: apps/patient/src/views/Main.tsx [hunk 6]
```diff
@@ -42,6 +47,8 @@ export const Main = () => {
     isDemo ? countFillsAndRemoveDuplicates(demoOrder.fills) : []
   );
 
+  const [totalWalmartPrice, setTotalWalmartPrice] = useState(0);
```
Comment (marisbest2) [inline apps/patient/src/views/Main.tsx:50]: useState<number | null>(null)

File: apps/patient/src/components/BrandedPharmacyCard.tsx [hunk 7]
```diff
@@ -46,8 +46,8 @@ export const BrandedPharmacyCard = ({ pharmacyId, handleSelect }: Props) => {
   ) : null;
 
   return (
-    <Container>
-      <Box py={2} bgColor="white" cursor="pointer" onClick={() => handleSelect?.(pharmacyId)}>
+    <Container p={3}>
+      <Box bgColor="white" cursor="pointer" onClick={() => handleSelect?.(pharmacyId)}>
```
Comment (marisbest2) [inline apps/patient/src/components/BrandedPharmacyCard.tsx:50]: What's with the style changes in this PR?

**Review by marisbest2 — DISMISSED** [review]

**Comment by marisbest2** [comment]
Do we want/need a disclaimer on the pricing?

File: apps/patient/src/components/BrandedPharmacyCard.tsx (hunk 7 above)
Comment (jknapp25) [reply apps/patient/src/components/BrandedPharmacyCard.tsx]: inconsistent padding on pickup vs branded pharmacies here

File: apps/patient/src/utils/general.ts (hunk 4 above)
Comment (jknapp25) [reply apps/patient/src/utils/general.ts:249]: first pass was keeping just number, but I'll update

File: apps/patient/src/utils/general.ts (hunk 3 above)
Comment (captDaylight) [reply apps/patient/src/utils/general.ts:230]: I think that might be hard to ensure an exact match and the example might be an edge case, IMO this should be good to go because this is at max a 2 or 3 week test that will affect 200 - 400 orders. I think getting this out today is more the higher priority since we have an upcoming meeting with Walmart, hoping to show some positive results

**Review by captDaylight — APPROVED** [review]
